### ISSUE TITLE
Cloud SQL mysql wont be connect from my laptop (using Cloud SQL Auth proxy)

### ISSUE THREAD

**Issue by aslamchandio** [GoogleCloudPlatform/cloud-sql-proxy#1818]
### Question

I created Cloud SQL (mysql) Instance using private IP Address (Google Service Access) ,i am unable to connect my sql instance from my laptop using Cloud SQL Auth proxy

./cloud-sql-proxy --address 127.0.0.1  --port 6000   (connection name) --private-ip --credentials-file  sakey.json(service account key)

Service Account have following permissions

Cloud SQL Admin
Cloud SQL Client
Cloud SQL Editor
Project IAM Admin (even i try with this)

when i connect with cloud sql 
mysql -u root --host 127.0.0.1 -P 6000  -p

Error :

ERROR 2013 (HY000): Lost connection to MySQL server at 'reading initial communication packet', system error: 2

ailed to connect to instance: Dial error: failed to dial (connection name = ""): dial tcp 10.88.0.2:3307: i/o timeout

**Note :* I am able to connect sql Instance when i created gce vm in peered network with cloud sql instance 

but my goal is to connect that instance from my laptop..

**Comment by jackwotherspoon** [comment]
Hi @aslamchandio ! Thanks for the question about the Cloud SQL Auth Proxy 😄 

When you create a Private IP Cloud SQL instance you attach to it a VPC private network. Only connections from this VPC network or a peered VPC network will be able to connect to the Private IP of the instance. This is how Private IP is meant to work and why it is more secure than connecting over Public IP.

The Cloud SQL Auth Proxy does not provide a new connectivity path; it relies on existing IP connectivity. To connect to a Cloud SQL instance using [private IP](https://cloud.google.com/sql/docs/mysql/private-ip), the Cloud SQL Auth Proxy must be on a resource with access to the same VPC network as the instance.

**TLDR**: Your local laptop is not connected to the VPC network and thus does not have a connectivity path to your Private IP instance. This is working as intended, connecting from your local computer would be over Public IP.

If your goal is to connect easily from your local computer to a Cloud SQL instance then you may want to add a Public IP Address to your instance, and allowlist your local computer's IP address to connect. Otherwise, an alternative would be having a GCE VM attached to the VPC network of the Cloud SQL instance and SSH to it from your local computer, [similar steps to as shown here](https://cloud.google.com/sql/docs/mysql/connect-instance-private-ip).

**Comment by aslamchandio** [comment]
Hi [jackwotherspoon](https://github.com/jackwotherspoon),
I achieved what i trying to do connect private sql instnace from my laptop using bellow link.
https://medium.com/google-cloud/cloud-sql-with-private-ip-only-the-good-the-bad-and-the-ugly-de4ac23ce98a
Thanks for your reply.

**Comment by jackwotherspoon** [comment]
Glad you achieved connectivity to your instance. I will close this as you resolved the question 😄
